### DISCUSSION TITLE
Fix cross-organisation "Allow edits from maintainers"

### DISCUSSION THREAD

**Discussion by xmo-odoo** [community/community#5634] [Pull Requests]
(aka "Allow edits by maintainers")

Currently, if a PR is created from a fork *living under an organisation* (rather than an individual) it's not possible to allow edits by maintainers. This is better than the previous state of affairs where the feature was available *but did not work*, but it remains less than great.

Cf  isaacs/github#1681 for a long list of maintainers mentioning exactly this issue on dozens of PRs (which probably undercounts the issue by orders of magnitude as they'd need to know about the informal isaacs tracker).

**Comment by joao-paulo-parity** [comment]
You might want to change the title to match the exact option name

```diff
-allow edit by maintainer
+Allow edits from maintainers
```

I just filed https://github.com/github/feedback/discussions/9921 because I was not able to find this issue before

**Reply by dafreitag** [reply]
Interestingly the [documentation ](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) shows a screenshot that is "Allow edits from maintainers", but I just did the same thing on a personal pull request and it shows "Allow edits by maintainers
![Unknown-3](https://user-images.githubusercontent.com/53196251/153501486-176c5603-97dc-4cd3-ae38-e5aa8a37f068.jpeg)
"

**Comment by joao-paulo-parity** [comment]
Moving over feedback from #9921

Personally, I do not see a reason for "Allow edits from maintainers" to have a different behavior if the pull request is coming from a organization. If you think otherwise and it's working as intended then please explicitly state what's the official decision regarding this behavior.

This is blocking some commit-automation-based tools from working as intended, as reported in https://github.com/isaacs/github/issues/1681#issuecomment-825453532

- aws/aws-cdk#14099
- paritytech/polkadot#2922

**Reply by xmo-odoo** [reply]
> Personally, I do not see a reason for "Allow edits from maintainers" to have a different behavior if the pull request is coming from a organization.

I would assume the reason is that organisation accounts have different and more complex collaboration models so not breaking anything is probably riskier, and furthermore orgs are commonly companies which may not want "unmanaged" / "stealth" contributors of that fashion.

Not that that makes this limitation any less of a pain, mind. But it's not hard to imagine why it would be.

**Comment by dafreitag** [comment]
This is increasingly becoming an issue for pull requests to outside projects from forks inside our Org.  I concur with @joao-paulo-parity  opinion.  If there is a technical limitation causing this option to be missing, we would like an explanation and clearer documentation as to the difference between Org and user created forks in respect to this option's availability.

Is there a simple work around in the meantime?  Can commits be added _directly_ to a pull request by the maintainer if they are added to our fork as an outside collaborator?  It's not an ideal solution if so, but may help with the issues we are experiencing while making contributions.

**Reply by xmo-odoo** [reply]
>  Can commits be added directly to a pull request by the maintainer if they are added to our fork as an outside collaborator?

Yes IIRC if the maintainer is added as a collaborator to the repository they'll get whatever access you grant them.

But that can be rather distasteful as the "write" access level (necessary for pushing code to the repository) also provides write access to a lot of meta-information which you may not want to grant to the maintainer on the other side. Though I guess it's less of an issue when it comes to forks.

**Comment by DenverCoder1** [comment]
This is still an issue.

I use [an organization account](https://github.com/DenverCoderOne) in order to keep my forked repos separate from my main projects, so I'm always making PRs from an organization account.

There have been several times I wanted to allow edits from maintainers, but was unable to find the option.

**Reply by danez** [reply]
Same exact use-case and problem here.

**Comment by luciosis** [comment]
IS there any update to this issue? we got same use-case and problem here

**Comment by ShayBox** [comment]
Same problem, nobody can commit to my forks, I keep them in an org for separation.

**Comment by denverbaumgartner** [comment]
still a problem :(

**Reply by akshatgurbuxani** [reply]
still a problem in 2025 :(((

**Comment by g-s-eire** [comment]
This is still an issue.

**Comment by dluc** [comment]
HELP! Please fix this, or we shall move to svn 🥲

**Comment by sethgw** [comment]
help

**Comment by piRGoif** [comment]
Hello,

We have the same issue in our Combodo/iTop repo. Some of our contributors are employees of companies that are Combodo partners, so they are using forks in their employer org and won't create forks in their personal account...
Sample PR where this was an issue : https://github.com/Combodo/iTop/pull/523#issuecomment-1672752425

I understand a repo in an organization is shared with more people than just personal repo... Maybe add org to org authorizations ? Like in the previous example, allow Combodo/iTop maintainers to have access to Itomig fork during a day ?

**Comment by FVidalCarneiro** [comment]
This seems to still be an issue and is limiting organization's ability to contribute to open source projects, example pull-request [here](https://github.com/airbytehq/airbyte/pull/34166).

**Comment by Superleah06** [comment]
Do not clone

**Comment by Superleah06** [comment]
[Revised-Form-U2_2017.pdf](https://github.com/community/community/files/14740988/Revised-Form-U2_2017.pdf)

Leah m. Guyer mon Mar 26,2024

**Comment by 3kh0** [comment]
Still a issue and waiting for something to be done about this

**Comment by dluc** [comment]
If "edit from maintainers" isn't an option for PR submitters, consider adding "block PRs from organizations" for repo owners. 

Receiving PRs without the ability to collaborate is frustrating, and repeatedly explaining the issue is a time-waster.

**Comment by ChrisBarker-NOAA** [comment]
WOW -- still an issue.

Anyway -- if it's not going to be changed, then at least put some prominent text on there, so I doin't spend who knows how long looking desperately for the check box!

Maybe, in place of the check box:

"Allowing edits from maintiners is not allows for org accounts' or SOMETHING!!

**Reply by rkboni** [reply]
Great idea, I just did the same thing this morning, looking at the PR from all aspects, looking through repo settings, etc.  Also, mentioning this in the docs would be a great idea!

**Reply by KS-HTK** [reply]
And add a edits are disabled warning to the PR so I don't spend time working on the PR only to realize I can't even push it…

**Comment by geky** [comment]
I think this is also affecting forks of org forks, even if the resulting fork don't live in an organization. Which is quite confusing for new users.

Example: https://github.com/littlefs-project/littlefs/pull/1013

**Comment by rkboni** [comment]
Another example of this: https://github.com/openwrt/openwrt/pull/16779#issuecomment-2447333401.  In my case I'm using an org because I have multiple forks of a project for collaborating across different feature sets (hardware in the case of OpenWRT), and I'm unable to keep multiple forks without an organization :/

**Comment by ccoVeille** [comment]
I also bumped on this after I made a script to clean up my forks

- https://github.com/ccoVeille/github-forks-sanitizer/issues/3

They have to understand and accept. Some of them decline to join my fork write write permission, but they complain about not being able to push on my fork.

EDIT: ~~For now, I plan to find repositories maintainers via the API, then onboard them on my forks with "write" permission, but it's a pain.~~ _after @dluc feedback_

So, yes I also need a solution for organization PR

**Reply by dluc** [reply]
> For now, I plan to find repositories maintainers via the API, then onboard them on my forks with "write" permission, but it's a pain.

@ccoVeille I'd be very cautious about that, write permission could lead to unexpected incidents, e.g. if your fork has billable resources or "unexpected" content in your repo could bring you legal problems. Consider also that you're trusting other repo maintainers not to have their account hacked, which could affect you too.   Check this table https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization

**Reply by ccoVeille** [reply]
Very wise warning. I will reconsider the way I do. I don't think I can onboard people automagically.

**Comment by woodruffw** [comment]
This is still an issue in 2025. Others have pointed out that there are architectural and security reasons why this wouldn't be trivial to do by default for organizations, which is very fair.

However, it would be *really nice* if this behavior was more clearly signaled on Pull Requests: I have 3-4 conversations a month on Pull Requests where the other maintainer is (reasonably) confused and frustrated because they can't make suggestions or push to my branch, and they typically assume it's a configuration problem (i.e. my fault) instead of GitHub's own limitation.

To that end, even just a small warning on the Pull Request's sidebar explaining that organization forks don't support suggestions or edits from upstream maintainers would go a long way.

**Reply by Wxh16144** [reply]
Agree, even if the design does not support it temporarily, the restricted situations should be marked otherwise it is too painful to explain repeatedly.

**Comment by maroshvarchola** [comment]
Hm interesting. I just crashed into this, and now we need to move our changes from org fork to my for, which takes little bit time, at least when you already create PR already... I mean, is there any solution, on lets say: I have org, we have team and we are using opensource, but, in opensource there is a bug, so our team will create fix (of course in our org fork as it makes sense?) and then what? Somebody then need to manually push that branch to someone's personal fork? I mean... this sounds stupid, or am I stupid?

**Reply by dalito** [reply]
An idea that could work is if GitHub would give org admins a way to assign an "allow edits" permission to certain trusted members.

**Reply by upsuper** [reply]
A workaround is to invite the upstream maintainers as collaborators of the specific repo with direct access. This would enable them pushing changes. But admittedly this is more cumbersome as they need to accept the invitation and you may need to add each individual separately.

**Comment by NorseGaud** [comment]
Bump

**Comment by bernalde** [comment]
This is still an issue https://github.com/Pyomo/pyomo/pull/3830

**Comment by ChampionX1001** [comment]
This is still an issue 
This needs to be fixed / allowed as an option, as my organization is trying to collaborate on an addition to an open source project. However, we can't make a PR to the open source project from our org repo without the maintainer asking for permission to edit the PR

**Comment by Fantu** [comment]
I also think that is needed.
Without it PRs from a repository within an organization, you risk wasting more time on small changes that maintainers could quickly make.
And if you don't waste more time on both sides (requesting the change and the other making the change), you'll have to do it later, risking unnecessary commits, and in some cases even unnecessary CI runs.
At the moment, the only workaround I can think of is for a contributor to create a personal fork and put the PR branch there (if it's also needed within the organization, it will take longer to keep it synchronized).
Please consider this addition; even if it's disabled by default and must be enabled in the organization's settings, it would be fine.
